compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp

```diff
@@ -1047,7 +1070,11 @@ bool ConvertNCHWToNHWCPass::run(loco::Graph *g)
       ConvertNCHWToNHWC converter;
       auto circle_node = loco::must_cast<luci::CircleNode *>(node);
       if (circle_node->rank() != 4)
-        continue;
+      {
+        // Q(kvochko) What will happen if we remove rank-4 check?
```

Comment:
I suppose it could generate wrong IR for some operations. (insert 4d transpose for non 4d tensors)

Reply:
I guess so. I'm only a bit concerned about this workaround, because we are starting to allow non-4d nodes in this pass. If more such nodes are introduced, this condition will grow and it could turn out that it's better to check assert that the rank of a node's input is 4, rather than of the node itself.

Reply:
Is it possible to transfrom NCHW -> NHWC in a non-rank 4 node?

For example, how can you identify the original layout of rank 3, 2, or 5?